src/rw.jl

```diff
@@ -0,0 +1,196 @@
+using ADIOS2
+
+    """
+    Intialize ADIOS2. Supports several configuration options: MPI, serial,
+    existing XML config file or a new file with specified parameters.
+    """
+    function adios2_init(; filename::AbstractString = "", engine_type = "",
+                         serial = false, comm = nothing)
+
+        use_mpi = !isnothing(comm)
+
+        adios = nothing
+
+        if isempty(filename)
+            if serial
+                global adios = adios_init_serial()
+            elseif use_mpi
+                filename = adios2_config(engine = engine_type)
+                global adios = ADIOS2.adios_init_mpi(joinpath(pwd(), filename), comm)
+            else serial_file
+                filename = adios2_config(engine = engine_type)
+                global adios = ADIOS2.adios_init_serial(joinpath(pwd(), filename))
+            end
+        else
+            if use_mpi
+                global adios = ADIOS2.adios_init_mpi(joinpath(pwd(), filename), comm)
+            else
+                global adios = ADIOS2.adios_init_serial(joinpath(pwd(), filename))
+            end
+        end
+
+        global init_state = true          # mark the initialization step
+        global vars = []                  # initialize the variable array (used later for reading and writing)
+        global nprocessed = 0            # needed for the reading step
+        global reading_now = false        # flag for the reading loop (to avoid race conditions)
+    end
+
+    """
+    Initialize io in read mode and the corresponding engine for reading data.
+    """
+    function read_setup(bp_filename = "")
```

Comment:
Change `bp_filename` to `filename` everywhere. BP is his only one of multiple file types ADIOS supports and in addition it support also the SST mode, which is one of the primary reasons why we are interested in ADIOS.

Reply:
Also `bp_path` to `path`